src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -485,6 +494,27 @@ protected async Task<ConnectNewHeadersResult> PresentHeadersLockedAsync(List<Blo
             return result;
         }
 
+        private bool CheckIfUnsolicitedFutureHeader(List<BlockHeader> headers)
+        {
+            if (headers.Count == 1 && this.BestReceivedTip != null)
+            {
+                // Distance of header from the peer expected tip.
+                double distanceSeconds = (headers[0].BlockTime - this.BestReceivedTip.Header.BlockTime).TotalSeconds;
+
+                if (this.chain.Network.MaxTipAge < distanceSeconds)
```

Comment:
The potential problem I see here is if a block is mined beyond `MaxTipAge ` the chain will be stuck.

Reply:
I'm not so sure, don't we have periodic timers that eventually force a `Resync` regardless?

In terms of the usages of this method, if we are unable to connect an unsolicited header then we are either in IBD or it is a header staked on a fork. If, in addition to being unconnectable, the header is far in the future, then staking has stalled for a very long time or we are in IBD.

Reply:
Ah correct if the headers are connectable we never reach this part so its a none issue.